pkg/component/controller/apiendpointreconciler.go

```diff
@@ -37,26 +38,28 @@ import (
 
 // Dummy checks so we catch easily if we miss some interface implementation
 var _ manager.Component = (*APIEndpointReconciler)(nil)
-var _ manager.Reconciler = (*APIEndpointReconciler)(nil)
 
 // APIEndpointReconciler is the component to reconcile in-cluster API address endpoint based from externalName
 type APIEndpointReconciler struct {
-	clusterConfig *v1beta1.ClusterConfig
-
 	logger *logrus.Entry
 
+	externalAddress   string
+	apiServerPort     int
 	leaderElector     leaderelector.Interface
-	stopCh            chan struct{}
 	kubeClientFactory kubeutil.ClientFactoryInterface
+
+	stopCh chan struct{}
 }
 
 // NewEndpointReconciler creates new endpoint reconciler
-func NewEndpointReconciler(leaderElector leaderelector.Interface, kubeClientFactory kubeutil.ClientFactoryInterface) *APIEndpointReconciler {
+func NewEndpointReconciler(nodeConfig *v1beta1.ClusterConfig, leaderElector leaderelector.Interface, kubeClientFactory kubeutil.ClientFactoryInterface) *APIEndpointReconciler {
```

Comment:
I'd suggest renaming it is not a reconciler anymore since there is no Reconcile method or interface.

Plus I think, more or less the same logic is in the TunneledEndpointReconciler which is still not dropped, good to check if it is required to use node config.

Reply:
> I'd suggest renaming it is not a reconciler anymore since there is no Reconcile method or interface.

It doesn't take dynamic configuration into account, but it still has the ten-second reconciliation loop in which it syncs any DNS changes to the endpoints object.

> Plus I think, more or less the same logic is in the TunneledEndpointReconciler which is still not dropped, good to check if it is required to use node config.

Indeed. The TunneledEndpointReconciler was looking at the dynamic config, too. Removed that part.